Add call to vpc-cni-k8s-connector for EKS windows

KlwntSingh opened the pull request:
CNI refactor project about removing k8s dependencies and using k8s-connector to get pod IP.

*Issue #, if available:*

*Description of changes:*
1. CNI plugin depends on vpc-cni-k8s-connector when using CNI in EKS Windows
2. Connector is invoked by CNI and returned IP CIDR string is processed.

*Tested*
Tested both connector and CNI on EKS Windows.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

rawahars commented on plugins/vpc-shared-eni/plugin/plugin.go:
```diff
@@ -30,7 +30,7 @@ const (
 
 var (
 	// specVersions is the set of CNI spec versions supported by this plugin.
-	specVersions = cniVersion.PluginSupports("0.3.0", "0.3.1")
```
Please just check if this is supported by Linux as well.

KlwntSingh replied:
Since CNI conf file will specify CNI plugin version to use, adding 0.4.0 will be backward compatible even if linux does not support it.

rawahars commented on plugins/vpc-shared-eni/config/k8s_connector.go:
```diff
@@ -0,0 +1,76 @@
+}
+
+// retrievePodIPWithK8sConnector retrieves Pod IP using k8s connector binary.
+func retrievePodIPWithK8sConnector(podName, podNamespace string) (string, error) {
```
nit: Maybe change the order to namespace and then name. It seems more intuitive i.e. `pod Namespace` -> `podName`.

rawahars commented on plugins/vpc-shared-eni/config/k8s_connector.go:
```diff
@@ -0,0 +1,76 @@
+
+	kc := netConfig.Kubernetes
+
+	// retrieve Pod IP Address CIDR string using podName, podNamespace
```
nit: add a full stop at the end of comment.

rawahars commented on plugins/vpc-shared-eni/config/k8s_connector.go:
```diff
@@ -0,0 +1,76 @@
+// retrievePodIPWithK8sConnector retrieves Pod IP using k8s connector binary.
+func retrievePodIPWithK8sConnector(podName, podNamespace string) (string, error) {
+	// Get Path to aws-vpc-cni-k8s-connector binary
+	connectorBinaryPath := os.Getenv("AWS_VPC_CNI_K8S_CONNECTOR_BINARY_PATH")
```
Add a comment about this env in the readme so that people know what is needed to be done for Windows.

rawahars commented on plugins/vpc-shared-eni/config/k8s_connector.go:
```diff
@@ -0,0 +1,76 @@
+
+	// Return error if exit code is other than 0
+	if cmd.ProcessState.ExitCode() != 0 {
+		return "", fmt.Errorf("retrievePodIPWithK8sConnector: failure in connector %s", errBytes.String())
```
nit: Try to make the error message format same across all the messages.

ofiliz committed:
Rename the vpc-shared-eni plugin to vpc-bridge

Renaming the files in preparation for vpc-bridge plugin implementation.

ofiliz committed:
Update vpc-bridge make target and package names

YashdalfTheGray committed:
Fix end to end tests and some other improvements (#90)

* satisfied the missing arguments for cni invoke.ExecPluginWithoutResult()

* run tests on every push, across all branches

* created a defaultExec in the file rather than depending on invoke's

* more potential test fixes

* added logs collection to the github action for when tests fail

* more fixes, fixed logs path in the e2e action

* switched to using context.Background because that makes more sense here

* preserve all logs in the e2e build

* make directories using os.mkdirtemp rather than ioutil

* changed the permissions on the log directory to pull out the logs

* fixed the permission because it's a uint and not a string

* commented out the test to see if something else is failing too

* started a testing.md file to contain some testing instructions

* updated the network namespace names to be sufficiently different

* updated the testing doc to have a vscode debugging profile for the e2e tests

* fixed ubuntu meta package name, added a pretest command, and names for all tasks

* added a couple more debug commands to the e2e workflow

* added some more details to the testing writeup

* added a copy of all the test resources for both the block imds version of the test and without

* use a singular template with a fields hash to differentiate between the tests

* added more details to the testing page, added a link to the testing page to contributing, formatted

* left a comment to provide users more context on the pretest step

* refactored the test code to pass the right args into the validation functions

rawahars committed:
Upgrade hcsshim, golang.org/x/sys and testify dependencies (#91)

* updated hcsshim to 0.9.4

Updated the version of hcsshim from `0.7.12` to `0.9.4`. Then ran `go mod tidy`

* updated the vendor directory for upgrading hcsshim to 0.9.4

Ran `go mod vendor` after the previous commit where we updated the hcsshim package to `0.9.4` and ran `go mod tidy`.

* updated golang.org/x/sys to `0.7.0` and github.com/stretchr/testify to `1.8.2`

Ran `go mod tidy` after updating the versions

* updated the vendor directory for upgrading stretchr/testify and sys packages

Ran `go mod vendor` after the previous commit where we updated the package versions.

KlwntSingh committed:
Add call to vpc-cni-k8s-connector for EKS windows

1. CNI plugin depends on vpc-cni-k8s-connector when using CNI in EKS Windows
2. Call is made to binary and returned output is processed.

KlwntSingh committed:
Add named pipe for reading output from vpc connector binary

1. Added creating named pipe.
2. Added listening and reading from named pipe.
